src/redux/Products/thunk.js

```diff
@@ -0,0 +1,91 @@
+import Swal from 'sweetalert2';
+import {
+    setProducts,
+    setLoadingProducts,
+    dismissLoadingProducts,
+    getProductsError,
+    deleteProductSuccess,
+    getProductDelete
+} from './actions'
+
+const BASE_URL = 'http://localhost:4000'
+
+export function getProducts() {
+    return async (dispatch) => {
+        dispatch( setLoadingProducts() );
+        try {
+            let response = await fetch(`${BASE_URL}/product`);
+            const responseParser = await response.json();
+            if(responseParser.data) {
+                dispatch(dismissLoadingProducts())
+                dispatch( setProducts(responseParser.data) )
+            }
+        } catch (error) {
+            console.log(error);
+            dispatch( getProductsError() )
+        }
+    }
+}
+
+export function deleteProduct(id) {
```

Comment:
same logic here of handling responses with status code, save your response in a constant, and check for the status